yt_dlp/extractor/megatvcom.py

```diff
@@ -0,0 +1,201 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+from urllib.parse import (
+    parse_qs,
+    urlparse,
+)
+
+from .common import InfoExtractor
+from ..utils import (
+    clean_html,
+    determine_ext,
+    ExtractorError,
+    extract_attributes,
+    get_element_by_class,
+    HEADRequest,
+    unescapeHTML,
+    unified_timestamp,
+)
+
+
+class MegaTVComBaseIE(InfoExtractor):
+    _PLAYER_DIV_ID = 'player_div_id'
+
+    def _extract_player_attrs(self, webpage):
+        PLAYER_DIV_RE = r'''(?x)
+        <div(?:
+            id=(?P<_q1>["'])(?P<%(pdi)s>%(pdi)s)(?P=_q1)|
+            [^>]*?
+        )+>
+        ''' % {'pdi': self._PLAYER_DIV_ID}
+        for mobj in re.finditer(PLAYER_DIV_RE, webpage):
+            if mobj.group(self._PLAYER_DIV_ID):
+                player_el = mobj.group(0)
+                break
+        else:
+            raise ExtractorError('no <div id="%s"> element found in webpage' %
+                                 self._PLAYER_DIV_ID)
+        return {
+            re.sub(r'^data-(?:kwik_)?', '', k): v
+            for k, v in extract_attributes(player_el).items()
+            if k not in ('id',)
+        }
+
+
+class MegaTVComIE(MegaTVComBaseIE):
+    IE_NAME = 'megatvcom'
+    IE_DESC = 'megatv.com videos'
+    _VALID_URL = r'https?://(?:www\.)?megatv\.com/(?:(?!\d{4})[^/]+/(?P<id>\d+)/[^/]+|\d{4}/\d{2}/\d{2}/.+)'
+
+    _TESTS = [{
+        'url': 'https://www.megatv.com/2021/10/23/egkainia-gia-ti-nea-skini-omega-tou-dimotikou-theatrou-peiraia/',
+        'md5': '6546a1a37fff0dd51c9dce5f490b7d7d',
+        'info_dict': {
+            'id': '520979',
+            'ext': 'mp4',
+            'title': 'md5:70eef71a9cd2c1ecff7ee428354dded2',
+            'description': 'md5:0209fa8d318128569c0d256a5c404db1',
+            'timestamp': 1634975747,
+            'upload_date': '20211023',
+        },
+    }, {
+        'url': 'https://www.megatv.com/tvshows/527800/epeisodio-65-12/',
+        'md5': 'cba2085d45c1abeb8e7e9b7e1d6c0072',
+        'info_dict': {
+            'id': '527800',
+            'ext': 'mp4',
+            'title': 'md5:fc322cb51f682eecfe2f54cd5ab3a157',
+            'description': 'md5:b2b7ed3690a78f2a0156eb790fdc00df',
+            'timestamp': 1636048859,
+            'upload_date': '20211104',
+        },
+    }]
+
+    def _match_article_id(self, webpage):
+        ART_RE = r'''(?x)
+        <article(?:
+            id=(?P<_q2>["'])Article_(?P<article>\d+)(?P=_q2)|
+            [^>]*?
+        )+>
+        '''
+        return self._search_regex(ART_RE, webpage, 'article_id',
+                                  group='article')
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        _is_article = video_id is None
+        webpage = self._download_webpage(url,
+                                         'N/A' if _is_article else
```

Comment:
As you can see the webpage is subsequently downloaded in order to extract the actual `video_id`, among other tasks.
So I understand your suggestion is to temporarily use the said slug in order to show something other than 'N/A' in this case, rather than to set `display_id` to this slug.
Please advise if I have missed your point though.

Reply:
> So I understand your suggestion is to temporarily use the said slug in order to show something other than 'N/A' in this case

Yes, that's important part

> rather than to set display_id to this slug.

I think it makes sense to set that as `display_id` in the info_dict too. But not mandatory